IaC/PnP/monitoring/aks/online/aks_secure_baseline/ngsa-memory/ngsa-memory.yaml

```diff
@@ -0,0 +1,122 @@
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    name: ngsa
+  name: ngsa
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ngsa-memory
+  namespace: ngsa
+# We should label our app with proper k8s label
+  labels:
+    app.kubernetes.io/name: ngsa-memory
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: ngsa-memory
+  template:
+    metadata:
+      labels:
+        app: ngsa-memory
+        version: beta-0329-1641
+        deploy: 21-03-29-16-41-59
+    spec:
+      containers:
+        - name: app
+          image: ghcr.io/retaildevcrews/ngsa-app:beta
+          imagePullPolicy: Always
+
+          args: 
+          - --in-memory
+          - --prometheus
+          # For now using dev zones 
+          # - --zone
+          # - az-central
+          # - --region
+          # - Central
+
+          ports:
+            - name: http
+              containerPort: 8080
+              protocol: TCP
+      # Added Node Selector
+      # Since we have two node pools for PnP cluster
+      nodeSelector:
+        agentpool: npuser01
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: ngsa-memory
+  namespace: ngsa
+spec:
+  type: ClusterIP
+  ports:
+    - port: 8080
+      targetPort: http
+      protocol: TCP
+      name: http
+  selector:
+    app: ngsa-memory
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: ngsa-memory-ingress
+  namespace: ngsa
+  annotations:
+    kubernetes.io/ingress.allow-http: "false"
+    # defines controller implementing this ingress resource: https://docs.microsoft.com/en-us/azure/dev-spaces/how-to/ingress-https-traefik
+    # ingress.class annotation is being deprecated in Kubernetes 1.18: https://kubernetes.io/docs/concepts/services-networking/ingress/#deprecated-annotation
+    # For backwards compatibility, when this annotation is set, precedence is given over the new field ingressClassName under spec.
+    kubernetes.io/ingress.class: traefik-internal
+    # kubernetes.io/ingress.class: traefik
+    traefik.ingress.kubernetes.io/router.entrypoints: websecure
+    traefik.ingress.kubernetes.io/router.tls: "true"
+    traefik.ingress.kubernetes.io/router.tls.options: default
+    traefik.ingress.kubernetes.io/router.middlewares: app-gateway-snet@file, gzip-compress@file, ngsa-stripprefix@file
```

Comment:
We should search the `app-gateway-snet` and `gzip-compress` middlewares to see if we need them

Reply:
These other middlewares are used by Traefik. See lines 137-139 in the traefik.yaml file

Reply:
I don't think we need them for NGSA in memory. I'll leave them in the traefik.yaml file